nuxeo-services/nuxeo-platform-filemanager-core-listener/src/main/java/org/nuxeo/ecm/platform/filemanager/core/listener/MimetypeIconUpdater.java

```diff
@@ -132,12 +138,17 @@ public void updateBlobProperty(DocumentModel doc, MimetypeRegistry mimetypeServi
         }
 
         Blob blob = dirtyProperty.getValue(Blob.class);
-        if (blob != null && (blob.getMimeType() == null || blob.getMimeType().equals(OCTET_STREAM_MT))) {
-            // update the mimetype (if not set) using the the mimetype registry
-            // service
-            blob = mimetypeService.updateMimetype(blob);
-            doc.setPropertyValue(fieldPath, (Serializable) blob);
+        if (blob != null) {
+            if (blob.getMimeType() == null || blob.getMimeType().equals(DEFAULT_MIMETYPE)) {
+                // update the mimetype (if not set) using the the mimetype registry
+                // service
+                blob = mimetypeService.updateMimetype(blob);
+                doc.setPropertyValue(fieldPath, (Serializable) blob);
+            } else if (!mimetypeService.isMimeTypeNormalized(blob.getMimeType())) {
+                mimetypeService.getNormalizedMimeType(blob.getMimeType()).ifPresent(blob::setMimeType);
```

Comment:
Maybe add a comment to explain (as in the previous `if` branch), like "normalize mime type (if not yet normalized)"